static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx

```diff
@@ -46,28 +49,60 @@ function SampleImages({groupId}: Props) {
 
   return (
     <ChartPanel title={showImages ? t('Largest Images') : undefined}>
-      {showImages ? (
-        <ImageWrapper>
-          {filteredResources.map(resource => {
-            return (
-              <ImageContainer
-                src={resource[SPAN_DESCRIPTION]}
-                showImage={isImagesEnabled}
-                fileName={getFileNameFromDescription(resource[SPAN_DESCRIPTION])}
-                size={resource[`measurements.${HTTP_RESPONSE_CONTENT_LENGTH}`]}
-                key={resource[SPAN_DESCRIPTION]}
-              />
-            );
-          })}
-        </ImageWrapper>
-      ) : (
-        // TODO - the selection to show only links should be persisted for awhile
-        <DisabledImages onClickShowLinks={() => setShowImages(true)} />
-      )}
+      <SampleImagesChartPanelBody
+        onClickShowLinks={() => setShowImages(true)}
+        images={filteredResources}
+        isLoadingImages={isLoadingImages}
+        isImagesEnabled={isImagesEnabled}
+        showImages={showImages}
+      />
     </ChartPanel>
   );
 }
 
+function SampleImagesChartPanelBody(props: {
+  images: ReturnType<typeof useIndexedResourcesQuery>['data'];
+  isImagesEnabled: boolean;
+  isLoadingImages: boolean;
+  showImages: boolean;
+  onClickShowLinks?: () => void;
+}) {
+  const {onClickShowLinks, images, isLoadingImages, showImages, isImagesEnabled} = props;
+
+  const hasImages = images.length > 0;
+
+  if (!showImages) {
+    return <DisabledImages onClickShowLinks={onClickShowLinks} />;
+  }
+  if (showImages && isLoadingImages) {
+    return <LoadingIndicator />;
+  }
+  if (showImages && !hasImages) {
+    Sentry.captureException(new Error('No sample images missing'));
```

Comment:
wait sorry I didn't see this sooner, @DominikB2014 `Sentry.captureException` needs to be in a `useEffect` call because it is a side effect of the render.

Reply:
I'll make the change!